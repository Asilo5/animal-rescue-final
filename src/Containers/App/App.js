import React, { Component } from 'react';
import { getAnimals, getDonations } from '../../apiCalls';
import { listedAnimals, receivedDonations } from '../../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Animals } from '../Animals/Animals';

export class App extends Component {
  async componentDidMount() {
      try {
          const resp = await getAnimals();
          console.log(resp)
          this.props.listedAnimals(resp);
          this.appendDonations();
      } catch(error) {
          console.log('Error getting animals', error);
      }
  } 
  
  appendDonations = async () => {
    try {
        const resp = await getDonations();
        this.props.receivedDonations(resp);
    } catch(error) {
        console.log('Error getting donations', error);
    }
  }

  render() {
    return (
        <main>
          <Animals />
        </main>
    )
  }
}

export const mapDispatchToProps = dispatch => (
    bindActionCreators({ 
        listedAnimals, 
        receivedDonations
    }, dispatch)
  );

export default connect(null, mapDispatchToProps)(App);
