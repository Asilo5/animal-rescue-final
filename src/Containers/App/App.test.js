import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
   let wrapper;
   let mocklistedAnimals = [
     {id: 1, name:'bob', species:'who knows', description:'who cares'}
   ];
   let mockDonations = [
     {name: 'bob', donation: 0}
   ]

   beforeEach(() => {
     wrapper = shallow(<App  
        listedAnimals={mocklistedAnimals}
        receivedDonations={mocklistedAnimals}
     />)
   })

   it('should match snapshot with all tests passing through correctly', () => {
     expect(wrapper).toMatchSnapshot();
   })

})
