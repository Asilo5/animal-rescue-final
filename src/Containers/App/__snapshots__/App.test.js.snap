// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with all tests passing through correctly 1`] = `undefined`;
